include/nil/blueprint/input_reader.hpp

```diff
@@ -455,11 +481,17 @@ namespace nil {
                     }
                 }
 
-                // Check if there are remaining elements of public input
-                if (function.arg_size() - ret_gap != public_input.size()) {
-                    error = "too many values in the input file";
+                // Check if there are remaining elements of input
+                if (function.arg_size() - ret_gap!= public_input.size() + private_input.size()) {
+                    log.debug(boost::format("public_input size: %1%") % public_input.size());
+                    log.debug(boost::format("private_input size: %1%") % private_input.size());
+                    log.debug(boost::format("ret_gap: %1%") % ret_gap);
+                    log.debug(boost::format("function.arg_size(): %1%") % function.arg_size());
+
+                    error = "too many values in the input files, public + private input sizes must be equal to function.arg_size - ret_gap";
```

Comment:
`ret_gap` is a detail of implementation (as well as `function.arg_size` btw), so it should not be exposed in error messages for a user (unlike asserts, that are intended to developers). Let's provide a readable error message that could help to fix a problem without examining the source code of assigner.